### DISCUSSION TITLE
Prop-driven behavior

### DISCUSSION THREAD

**Discussion by bombillazo** [gorhom/react-native-bottom-sheet#1171] [General]
Hello, is there any plans or way to control the BottomSheet component from a parent via props? Trying to use imperative functions ala class components in React Native functional components to open, close, snap to location, etc feels like an anti-pattern and breaks the convention of most of our internal component libraries. Beyond that our bottom sheet is hard to control and many times ends up in intermediate states that make it unusable until the app is refreshed.

Having the BottomSheet controls abstracted and the component takes care of the state given the props is a much better way to use the component. Just wanted to know if I was missing something or at least guidance on how to use the component properly without relying on `useEffect` hooks to monitor and sync the component open/close/snapping states. Thanks!

**Comment by kitze** [comment]
ping @gorhom, implementing  this manually is a huge pain :(

**Comment by anton-patrushev** [comment]
I guess it's all coming from that `BottomSheet` is always rendered and mounted and it's content as well. Ideally we should think about the `BottomSheet` as any other modal component which mounts its content only once it is `isVisible={true}` and unmounts once its `isVisible={false}`.

Today it leads to having N bottom sheet components rendered (if you have N different bottom sheets) on the same surface/screen  even if you need only 1 per time.

Speaking about native iOS realm — [the concept of the `bottom sheet` is just following](https://developer.apple.com/design/human-interface-guidelines/sheets) the same [Modality concept as any other modal-like view](https://developer.apple.com/design/human-interface-guidelines/modality), but it's just a different type of a presentation.

And native iOS sheets implements the modal like UI on the bigger windows/screens view ports out-of-box, which if we move towards the `isVisible` props driven design should be more user-friendly for react developers, given that this is community-wise pattern to render Modals through a declarative API, rather then always mount them (keep in the memory, track they rendering life cycle, etc.) and then call the `.present()`-like API on demand.

There was a separated conversion about that specifically, but this sounds to be somehow related as well. 
- https://github.com/gorhom/react-native-bottom-sheet/discussions/234

Would love to see your thoughts on it @gorhom

**Comment by CommanderRedYT** [comment]
I would also very much like a `open` prop or something similar, I am currently struggling to implement this in my app correctly.
